onyo/lib/tests/test_onyo.py

```diff
@@ -38,10 +38,129 @@ def test_OnyoRepo_incorrect_input_arguments_raise_error(repo: OnyoRepo,
+    length, and a body with the paths to changed files and directories relative
+    to the root of the repository.
+    """
+    modified = [repo.git.root / 's p a c e s',
+                repo.git.root / 'a/new/folder']
+
+    # modify the repository with some different commands:
+    mkdir(repo, modified, quiet=True, yes=True, message=None)
+    mv(repo, *modified, quiet=True, yes=True, message=None)
+
+    # generate a commit message:
+    message = repo.generate_commit_message(cmd='TST', modified=modified)
+    header = message.split("\n")[0]
+    list_files_changed = message.split("\n\n")[-1]
+
+    # root should not be in output
+    assert str(repo.git.root) not in message
+
+    # verify all necessary information is in the header:
+    assert f'TST [{len(modified)}]: ' in header
+
+    # verify all necessary information is in the body:
+    assert 'a/new/folder' in list_files_changed
+    assert 's p a c e s' in list_files_changed
+    assert len(list_files_changed.split("\n")) == len(modified)
+
+
+@pytest.mark.repo_files('a/test/asset_for_test.0')
+def test_is_onyo_path(repo: OnyoRepo) -> None:
+    """
+    Verify that `OnyoRepo.is_onyo_path()` differentiates correctly between
+    paths under `.onyo/` and outside of it.
+    """
+    # True for the directory `.onyo/` itself
+    assert repo.is_onyo_path(repo.dot_onyo)
+    # True for the directory `templates` inside of `.onyo/`
+    assert repo.is_onyo_path(repo.dot_onyo / 'templates')
+    # True for a file inside `.onyo/`
+    assert repo.is_onyo_path(repo.dot_onyo / 'templates' / 'empty')
+
+    # False for root of onyo repository
+    assert not repo.is_onyo_path(repo.git.root)
+    # False for directory `.git/`
+    assert not repo.is_onyo_path(repo.git.root / '.git')
+    # False for directory inside an onyo repository
+    assert not repo.is_onyo_path(repo.git.root / 'a' / 'test')
+    # False for asset inside an onyo repository
+    assert not repo.is_onyo_path(repo.git.root / 'a' / 'test' / 'asset_for_test.0')
+
+
+def test_Repo_get_template_file(repo: OnyoRepo) -> None:
+    """
+    The function `OnyoRepo.get_template_file()` must return a Path for all
+    file names of templates in `.onyo/templates/*`, and return the default
+    template if called without a file name specified.
+    """
+    # Call the function without parameter to get the default template `empty`:
+    assert repo.get_template_file().samefile(repo.git.root / '.onyo' /
+                                             'templates' / 'empty')
+
+    # from the templates dir, use the filename of each template to find the
+    # corresponding template file as a path.
+    for path in (repo.git.root / '.onyo' / 'templates').iterdir():
+        if path.name == '.anchor':
+            continue
+
+        template = repo.get_template_file(path.name)
+        assert isinstance(template, Path)
+        assert template.is_file()
+        assert '.onyo' in template.parts and 'templates' in template.parts
```

Comment:
```
 Path.samefile(other_path)

    Return whether this path points to the same file as other_path,
    which can be either a Path object, or a string.
```

We require it to return a `Path`, so I think I should not just remove the three lines for the usage of `samefile()` like suggested.
Otherwise thanks, I will add the other changes!